### PR TITLE
Validator registry logs cache

### PR DESCRIPTION
**Pull request by verkhohliad**

### CODE REVIEW

File: src/shared/ShutterTimer/useGetShutterValidatorIndexes.ts [hunk 1]
```diff
@@ -20,14 +20,21 @@ function extractValidatorIndex(messageHex: string) {
...
   const { data: logs } = useGetValidatorRegistryLogs(chainId);
 
   return useMemo(() => {
     return logs?.reduce((acc, log) => {
       const validatorIndex = extractValidatorIndex(log.args.message);
+      const subscriptionStatus = extractSubscriptionStatus(log.args.message);
 
-      acc.add(validatorIndex);
+      if (subscriptionStatus) {
+        acc.add(validatorIndex);
```
Comment (Wagalidoom) [inline src/shared/ShutterTimer/useGetShutterValidatorIndexes.ts:36]: We also need to delete validatorsIndex from acc in the case that its register at block n and deregister (status === '0') at block n + m

File: src/shared/ShutterTimer/useGetValidatorRegistryLogs.ts [hunk 2]
```diff
@@ -14,20 +14,31 @@ export const useGetValidatorRegistryLogs = (chainId: number) => useQuery({
       const rpc = chain.rpcUrls.default.http[0];
       const provider = new providers.JsonRpcProvider(rpc);
 
+      const localStorageLogsKey = [LOGS_QUERY_KEY, chainId].join('_');
+      const cachedString = localStorage.getItem(localStorageLogsKey);
```
Comment (Wagalidoom) [inline src/shared/ShutterTimer/useGetValidatorRegistryLogs.ts:18]: I think we can store past fetched logs in a json or something to keep it accessible to every users accross all sessions?

**Review by ilge-ustun — DISMISSED** [review]
Looks good!

File: src/shared/ShutterTimer/useGetValidatorRegistryLogs.ts (hunk 2 above)
Comment (verkhohliad) [reply src/shared/ShutterTimer/useGetValidatorRegistryLogs.ts:18]: local storage is not possible to be shared to different sessions on other devices or other users

**Review by Wagalidoom — APPROVED** [review]

**Pull request merged**